pandas/core/generic.py

```diff
@@ -635,6 +635,86 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
         d[axis] = mapper
         return self.rename(**d)
 
+    def set_axis_name(self, name=None, level_names=None, axis=0,
```

Comment:
This means you won’t be able to set a `MultiIndex.name` when you have a `MultiIndex`. I’m actually alright with that now that I think about it more.


> On Jan 7, 2016, at 6:41 AM, Jeff Reback <notifications@github.com> wrote:
> 
> In pandas/core/generic.py <https://github.com/pydata/pandas/pull/11980#discussion_r49068986>:
> 
> > @@ -635,6 +635,86 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
> >          d[axis] = mapper
> >          return self.rename(**d)
> >  
> > +    def set_axis_name(self, name=None, level_names=None, axis=0,
> just accept name, very very confusing to accept 2 args here
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/pydata/pandas/pull/11980/files#r49068986>.
>